### PR TITLE
Integration of Apache Superset for Enhanced Data Visualization

### PR DESCRIPTION
**Pull request by IvanPedroza**
Description:

This pull request introduces a significant enhancement to our data visualization capabilities by integrating Apache Superset into our project.

Key Features Added:

**Superset Dashboard Embedding**: We have embedded Superset dashboards that can be accessed directly from a local host web browser.
**Dynamic Data Source Configuration**: Implemented functionality to configure data directly from the code.

Benefits:

**Enhanced Data Visualization**: With Superset's robust visualization tools for users to create, explore, and share interactive dashboards and reports.

## Summary by CodeRabbit

- **New Features**
  - Improved processing orchestration for watch and search history in the main application.
  - Introduced functionality for database interactions, data loading, and API authentication in Superset.
  - Added capabilities to create and update charts, dashboards, and datasets in Superset.

- **Enhancements**
  - Added a flag to skip preprocessing for quicker testing and future command line support.

- **Documentation**
  - Updated comments across various files to enhance clarity and future development.

- **Style**
  - Minor stylistic adjustments like additional blank lines in code files for better readability.

### CODE REVIEW

File: .DS_Store
Comment (coltonrobbins73) [inline .DS_Store:1]: .gitignore

File: data/.DS_Store
Comment (coltonrobbins73) [inline data/.DS_Store:1]: .gitignore

File: data/output/dash_ready_watch_data.csv
Comment (coltonrobbins73) [inline data/output/dash_ready_watch_data.csv:1]: .gitignore

File: data/output/extracted_watch_history.csv
Comment (coltonrobbins73) [inline data/output/extracted_watch_history.csv:1]: .gitignore

File: data/watch-history.json
Comment (coltonrobbins73) [inline data/watch-history.json:1]: .gitignore

**Review by coltonrobbins73 — CHANGES_REQUESTED** [review]
There are some personal data files that need to be gitignored. There are also some security risks with hard coded secrets in the source code. 

There are a few suggestions from coderabbitai as well. For public repos, coderabbit is free so it might be useful for you in the future.

**Comment by IvanPedroza** [comment]
@coltonrobbins73 I resolved the suggested changes by removing unnecessary files and lines of code. The functions will remain hardcoded until it is clear what parameters you'll want to pass them.

**Pull request merged**